dev/bootstrap

```diff
@@ -1,30 +1,48 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-SUPPORT_FIRECLOUD_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+export SUPPORT_FIRECLOUD_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 
-# Reset and upgrade SF
 cd ${SUPPORT_FIRECLOUD_DIR}
-git reset
-git stash --all
-git checkout master
-git fetch
-git reset --hard origin/master
-git clean -xdf .
+
+# Resetting the support_firecloud directory
+# if the git repo is on master and only on master.
+# A development branch will therefore not be reset.
+BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
+if [ "${BRANCH_NAME}" = "master" ]
+then
+    git reset
+    git stash --all
+    git checkout master
+    git fetch
+    git reset --hard origin/master
+    git clean -xdf .
+fi
 
 # Bootstrap OS
 OS=$(uname | tr "[A-Z]" "[a-z]")
 
-sudo true || {
-    >&2 echo "[ERROR] You need local admin rights to properly bootstrap." \
-    "If you don't have local admin rights, try contacting the IT department."
-    exit 1
-}
+# Check sudo permissions if using sudo.
+if [ "${SF_USE_SUDO:-true}" = "true" ]
+then
+    echo "Enter your sudo password for bootstrap."
+    sudo true
```

Comment:
this is not going to work I'm afraid
as we run "bash strict mode", meaning this http://redsymbol.net/articles/unofficial-bash-strict-mode/ except we leave `IFS` alone

so if `sudo true` fails, it will just shortcircuit the entire script

which is why you see this pattern of
```
this || that
```

which at times evolves into
```
this || {
  that
  and_that
}
```

and at times evolves into the less unreadable version 🙈 (which ofc could be rewritten to an `if`)
```
!this || !this_either || {
  that
  and that
}
```